src/noise/mod.rs

```diff
@@ -382,24 +382,41 @@ impl Tunn {
         Ok(TunnResult::Done)
     }
 
+    // Update the index of the currently used session, if needed
+    fn set_current_session(&self, new_idx: usize) {
+        let cur_idx = self.current.load(Ordering::Relaxed);
```

Comment:
I'm concerned about widespread use of `Relaxed` which gives no ordering guarantees. Especially that on x86 the hardware gives it more predictable behavior than the spec promises, so it may doubly blow up on ARM.

Unless ability to change the order of atomic operations is essential for performance, I'd prefer all such uses to use `SeqCst` just to be safe that they do what they seem to do.

Reply:
I am aware of the memory model. We really don't care about the exact ordering of counters here, as long as they move forward in general.

Reply:
And on ARM it matters for performance